sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -741,7 +741,7 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
 
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
-  def warehousePath: String = new Path(getConf(WAREHOUSE_PATH)).toString
+  def warehousePath: String = Utils.resolveURI(getConf(WAREHOUSE_PATH)).toString
```

Comment:
i agree with @vanzin about dislike for resolveURI. i expect paths without schemes to on my default filesystem.